saskatoon/harvest/api.py

```diff
@@ -220,3 +224,185 @@ def list(self, request, *args, **kwargs):
                                  'title': _("New Person")
                                  }
                          })
+
+
+class StatsView(LoginRequiredMixin, generics.GenericAPIView):
+    template_name = "app/stats.html"
+    queryset = Harvest.objects.filter(status="Succeeded")
+    lookup_url_field = "season"
+    lookup_field = "season"
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = HarvestFilter
+    filterset_fields = ('status', 'season')
+    permission_classes = [IsCoreOrAdmin]
+
+    def get(self, request, format="html", *args, **kwargs):
+        """returns statistics on harvests for all seasons or a specific season"""
+
+        season = self.kwargs.get("season", "all")
+        if season not in self.seasons:
+            raise Http404("Season Not found")
+
+        if season == "all":
+            self.harvest_queryset = self.filter_queryset(self.get_queryset())
+            self.harvest_yield_queryset = HarvestYield.objects.filter(harvest__in=self.harvest_queryset)
+
+        else:
+            self.harvest_queryset = self.filter_queryset(self.get_queryset()).filter(
```

Comment:
What I had in mind would remove the need for passing a URL parameter manually (i.e. you would use the `season` filter from HarvestFilter REST would then handle the url parameter `?season=<choice>` for us.). Using `ListAPIView` instead of `GenericAPIView` might actually be a good idea!
Basically `self.filter_queryset(self.get_queryset())` would already contain the information about the season (all successful harvests if the filter is not applied). 
I think what would help you to troubleshoot would be to the the REST filter form (which you probably don't access to because you have already defined the html `template_name`.